src/patches/mapkey.cc

```diff
@@ -0,0 +1,143 @@
+#include "config.h"
+#include "prime/EventSystem.h"
+#include "utils.h"
+#include <absl/strings/ascii.h>
+#include <absl/strings/str_split.h>
+#include <cstdint>
+#include <iostream>
+#include <key.h>
+#include <map>
+#include <mapkey.h>
+#include <modifierkey.h>
+#include <prime/TMP_InputField.h>
+#include <string>
+#include <string_view>
+
+MapKey::MapKey()
+{
+  this->Key = KeyCode::None;
+}
+
+MapKey* MapKey::Parse(std::string_view key)
+{
+  auto strippedKey = absl::StripTrailingAsciiWhitespace(key);
+  auto lowerKey    = absl::AsciiStrToUpper(strippedKey);
+  auto wantedKeys  = absl::StrSplit(lowerKey, "-", absl::SkipWhitespace());
+
+  auto mapKey = new MapKey();
+  for (auto wantedKey : wantedKeys) {
+    auto modifier = ModifierKey::Parse(wantedKey);
+    if (modifier) {
+      mapKey->Modifiers.emplace_back(modifier);
+      mapKey->Shortcuts.emplace_back(wantedKey);
+    } else {
+      auto parsedKey = Key::Parse(wantedKey);
+
+      if (Key::IsModifier(parsedKey)) {
+        continue;
+      }
+
+      if (parsedKey != KeyCode::None) {
+        mapKey->Key = parsedKey;
+        mapKey->Shortcuts.emplace_back(wantedKey);
+      }
+    }
+  }
+
+  if (mapKey->Key != KeyCode::None) {
+    return mapKey;
+  }
+
+  return nullptr;
+}
+
+bool MapKey::HasGameFunction(GameFunction gameFunction)
```

Comment:
As per my comment on `mapkey.h` a lot of these will have to change

Reply:
Revamped where possible.  There's now a single gamefunction array rather than a map.